### PR TITLE
Port baddbmm CUDA from TH to ATen

### LINKED ISSUE #24539: Migrate `baddbmm` and `baddbmm_` from the TH to Aten (CUDA)
Porting TH operators is essential for code simplicity and performance reasons.

Porting guides and Q&A are available in umbrella issue: #24507

Feel free to add @VitalyFedyunin as a reviewer to get a prioritized review.

cc @ezyang @gchanan @zou3519 @vincentqb @vishwakftw @jianyuh @nikitaved @pearu @mruberry @heitorschueroff

### PR DESCRIPTION
**Pull request by kurtamohler**
Fixes #24539

### CODE REVIEW

**Comment by ailzhang** [comment]
@VitalyFedyunin would you mind taking a look at this PR? (feel free to reassign if it doesn't match ;) Thanks!

**Comment by kurtamohler** [comment]
@VitalyFedyunin , just wanted to check, is this on your radar?

**Review by sab-regime — APPROVED** [review]

**Review by sab-regime — APPROVED** [review]

File: aten/src/ATen/native/cuda/LinearAlgebra.cu [hunk 1]
```diff
@@ -6,24 +6,21 @@
...
 
 Tensor& baddbmm__cuda(Tensor& self, const Tensor& batch1, const Tensor& batch2, Scalar beta, Scalar alpha) {
   return baddbmm_out_cuda(self, self, batch1, batch2, beta, alpha);
 }
 
-Tensor& bmm_out_cuda(Tensor &result, const Tensor& batch1, const Tensor& batch2) {
-  result.resize_({ batch1.size(0), batch1.size(1), batch2.size(2) });
-  return legacy::cuda::_th_bmm_out(result, batch1, batch2);
+Tensor& bmm_out_cuda(Tensor &result, const Tensor& self, const Tensor& mat2) {
+  result.resize_({ self.size(0), self.size(1), mat2.size(2) });
```
Comment (mruberry) [inline aten/src/ATen/native/cuda/LinearAlgebra.cu:18]: Use resize_output here. 

https://github.com/pytorch/pytorch/blob/dae94ed02299cd1e9840ce6e49ea142b4d118b2d/aten/src/ATen/native/Resize.cpp#L9

**Comment by zou3519** [comment]
Is this and https://github.com/pytorch/pytorch/pull/42553 doing the same thing? cc @anjali411

**Comment by kurtamohler** [comment]
> Is this and #42553 doing the same thing? cc @anjali411

Yes, @ngimel pointed this out as well. Unfortunately, I didn't see #42553 when I started working on this, because there was no link to it in the issue. Closing this PR and reassigning the issue.

**Pull request closed without merging**